server/impl/src/main/java/com/walmartlabs/concord/server/user/UserManager.java

```diff
@@ -88,6 +88,10 @@ public UserManager(UserDao userDao, TeamDao teamDao, AuditLog auditLog, List<Use
             }
         }
 
+        if(!provider.validate(username, userDomain)){
```

Comment:
Not sure if I understand the point of this method.
It will result in an additional `UserInfoProvider#getInfo` call [right after we did the first one](https://github.com/walmartlabs/concord/blob/b3e8438fd8f5320d9e16c203b8e64fc18c57428a/server/impl/src/main/java/com/walmartlabs/concord/server/user/UserManager.java#L83-L84)?

Reply:
Case for creating new user i,e when user doesn't exist in DB
[LocalUserInfoProvider#getInfo](https://github.com/walmartlabs/concord/blob/4b5a32810a3413e6b4edf382545389c40ec4e34e/server/impl/src/main/java/com/walmartlabs/concord/server/security/internal/LocalUserInfoProvider.java#L51) should give null and [LDAPUserInfoProvider#getInfo](https://github.com/walmartlabs/concord/blob/4b5a32810a3413e6b4edf382545389c40ec4e34e/server/impl/src/main/java/com/walmartlabs/concord/server/security/ldap/LdapUserInfoProvider.java#L64) should give UserInfo for creating a new user.

Since I have removed the secondary validation [#getinfo ](https://github.com/walmartlabs/concord/blob/4b5a32810a3413e6b4edf382545389c40ec4e34e/server/impl/src/main/java/com/walmartlabs/concord/server/security/ldap/LdapUserInfoProvider.java#L82) used to happen in create method, extra `UserInfoProvider#validate` method is enforced. Please let me know if there is any better way to handle this?

Reply:
I don't understand what we're validating here:
- we call `UserInfoProvider#getInfo` [here](https://github.com/walmartlabs/concord/blob/b3e8438fd8f5320d9e16c203b8e64fc18c57428a/server/impl/src/main/java/com/walmartlabs/concord/server/user/UserManager.java#L83);
- then we call `UserInfoProvider#validate` [here](https://github.com/walmartlabs/concord/blob/b3e8438fd8f5320d9e16c203b8e64fc18c57428a/server/impl/src/main/java/com/walmartlabs/concord/server/user/UserManager.java#L91);
- `UserInfoProvider#validate` has two implementations - one does nothing and another calls the same `UserInfoProvider#getInfo` method.

Another place where we call `UserInfoProvider#validate` is this [UserManager code](https://github.com/walmartlabs/concord/pull/327/files#diff-0aa5ae1e2a9388a41e0ab9adca08094ab4e5a0ca5ca2c9af845a9a1949cb0164R215).
See my comment there.

Am I missing something or validation does nothing and can be effectively replaced with `UserManager#getOrCreate` in `UserResource`?

Reply:
ok. I refactored a bit. removed `UserInfoProvider#validate` and placed validation in `UserManager#getOrCreate`. Main requirement is to `remove ldap calls when user is created from sso login`

Reply:
:+1: